stats/stats/src/update_group.rs

```diff
@@ -0,0 +1,523 @@
+/// > **A** ➡ **B**\
+/// Group triggres `A`, which triggers `B`. Everything is fine.
+/// - `A` is on, `B` is off:\
+/// > **A** ➡ **B**\
+/// Group triggres `A`, which triggers `B`. Everything is fine.
+/// - `A` is off, `B` is on:\
+/// > A ⇨ B\
+/// Group only contains `A`, which means nothing is triggered. Quite counter-intuitive
+/// (`B` is not triggered even though it's enabled).
+/// - `A` is off, `B` is off:\
+/// > A ⇨ B\
+/// Nothing happens, as expected.
+///
+/// Therefore, to make working with updates easier, it is highly recommended to include all
+/// dependencies into the group. Later this check might be included into the macro.
+///
+/// Similarly, it makes sense to include all dependants (unless they have some other heavy dependencies),
+/// since it should be very lightweight to update them together.
+///
+/// ## Reasoning for macro
+///
+/// Since group member types are different (and trait impls have different associated types),
+/// we can't use homogeneous collections like `Vec`.
+///
+/// Therefore, the macro helps to avoid boilerplate when defining the update groups.
+///
+/// ## Example
+///
+/// ```rust
+/// # use stats::{Chart, Named, construct_update_group, DateValueString, UpdateError};
+/// # use stats::data_source::{
+/// #     kinds::{
+/// #         updateable_chart::clone::{CloneChart, CloneChartWrapper},
+/// #         remote::{RemoteSource, RemoteSourceWrapper},
+/// #     },
+/// #     types::{UpdateContext, UpdateParameters},
+/// # };
+/// # use chrono::NaiveDate;
+/// # use entity::sea_orm_active_enums::ChartType;
+/// # use std::ops::RangeInclusive;
+/// # use sea_orm::prelude::DateTimeUtc;
+///
+/// struct DummyRemoteSource;
+///
+/// impl RemoteSource for DummyRemoteSource {
+///     type Point = DateValueString;
+///
+///     fn get_query(range: Option<RangeInclusive<DateTimeUtc>>) -> sea_orm::Statement {
+///         // not called
+///         unimplemented!()
+///     }
+///     async fn query_data(
+///         cx: &UpdateContext<'_>,
+///         range: Option<RangeInclusive<DateTimeUtc>>,
+///     ) -> Result<Vec<Self::Point>, UpdateError> {
+///         Ok(vec![])
+///     }
+/// }
+///
+/// struct DummyChart;
+///
+/// impl Named for DummyChart {
+///     const NAME: &'static str = "dummyChart";
+/// }
+/// impl Chart for DummyChart {
+///     fn chart_type() -> ChartType {
+///         ChartType::Line
+///     }
+/// }
+///
+/// impl CloneChart for DummyChart {
+///     type Dependency = RemoteSourceWrapper<DummyRemoteSource>;
+/// }
+///
+/// // use wrappers to utilize existing implementation of `DataSource`
+/// // `Chart` impl is delegated to `DummyChart`.
+/// type DummyChartSource =
+///     CloneChartWrapper<DummyChart>;
+///
+/// construct_update_group!(ExampleUpdateGroup {
+///     name: "exampleGroup",
+///     charts: [DummyChartSource],
+/// });
+/// ```
+///
+#[macro_export]
+macro_rules! construct_update_group {
+    ($group_name:ident {
+        name: $name:literal,
```

Comment:
could be obtained from `$group_name` using https://github.com/dtolnay/paste?tab=readme-ov-file#case-conversion

Reply:
decided to not change the case for easier comprehension of logs

Reply:
still removed the name field